app/chat_endpoint_handlers.py

```
@@ -298,6 +305,78 @@ async def chat_message_internal(chat_id: str, user_id: str, action: Action, back
+    try:
+        characters = list(mongo[mongodb_name(user_id, chat_id)]["characters"].find())
+    except Exception as e:
+        log_exception(e, "chat_endpoint_handlers.chat_message_internal")
+    messages = [{
+        "role": "system",
+        "content": ""
+    }]
+    sql_connection.ping()
+    cursor = sql_connection.cursor()
+    cursor.execute(
+        f"SELECT * FROM (SELECT creator, content, aid FROM `{mariadb_name(user_id, chat_id)}`.messages ORDER BY aid DESC LIMIT 20) as a ORDER BY aid;")
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)